tests/src/common/CMakeLists.txt

```diff
@@ -25,6 +25,25 @@ target_link_libraries(json_test PUBLIC common_json main_test gmock)
 gtest_discover_tests(json_test NO_PRETTY_VALUES)
 add_dependencies(tests json_test)
 
+add_executable(yaml_test EXCLUDE_FROM_ALL yaml_test.cpp)
+target_link_libraries(yaml_test PUBLIC common_yaml main_test gmock)
+
+add_test(
+  NAME
+    yaml_build_and_run_test
+  COMMAND
+    ${CMAKE_CTEST_COMMAND}
+      --build-and-test ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR}/build_yaml
+      --build-generator ${CMAKE_GENERATOR}
+      --build-makeprogram ${CMAKE_MAKE_PROGRAM}
+      --build-target yaml_test
+      --build-options
+        -DMENDER_USE_YAML_CPP=ON
+        -B ${CMAKE_BINARY_DIR}/build_yaml
+      --test-command ./tests/src/common/yaml_test
```

Comment:
Haha, I see what you mean! 😂 But kudos for finding it out!

I would recommend a comment here to explain why this monstrosity is here. I suppose it's sort of understandable because of the `MENDER_USE_YAML_CPP=ON`, but still.

Reply:
I bet there are also ways for making this smoother, but this is the only combo I got to work.

I would expect `--build-and-test` to be able to set a different build dir and obey by it - but it doesnt...